git_machete/github.py

```diff
@@ -180,8 +180,10 @@ def __fire_github_api_request(method: str, path: str, token: Optional[str], requ
                                  'Visit `https://github.com/settings/tokens` to generate a new one.')
         elif err.code == http.HTTPStatus.NOT_FOUND:
             raise MacheteException(
-                f'Given endpoint: {url}, not found in GitHub or you don\'t have'
-                f' the permission to access it (expired token?).')  # TODO (#164): make dedicated exception here
+                f'Given endpoint: {url}, not found in GitHub. You don\'t have the permission to access it '
+                f'or the given API request, which is based on {method} HTTP request, requires valid GitHub API token. \n'
+                f'You can provide the token via one of the: {get_github_token_possible_providers()} '
```

Comment:
For consistency with the above, I like that version more:
```suggestion
                f'Provide a GitHub API token with `repo` access via one of the: {get_github_token_possible_providers()} '
```

Reply:
(see above tho)